packages/tpetra/core/test/inout/MatrixMarket_Tpetra_CrsMatrix_Dist_Binary.cpp

```diff
@@ -0,0 +1,822 @@
+    for (int n = 0; n < nMatvecs; n++) {
+      Teuchos::TimeMonitor tt(*timer);
+      matrix.apply(xvec, yvec, Teuchos::NO_TRANS, 3., 2.);
+    }
+
+    // Import result to a vector with default Tpetra map for easy comparisons
+    Teuchos::RCP<map_t> defMap = 
+             rcp(new map_t(yvec.getGlobalLength(),0,yvec.getMap()->getComm()));
+    Teuchos::RCP<vector_t> ydef = 
+             rcp(new vector_t(defMap, yvec.getNumVectors()));
+    Tpetra::Export<> exportDef(yvec.getMap(), defMap);
+    ydef->doExport(yvec, exportDef, Tpetra::INSERT);
+
+    return ydef;
+  }
+
+  
+  //////////////////////////////
+  //  Compare computed norms to the baseline norms
+  int compareToBaseline(
+    const std::string testname, 
+    const Teuchos::RCP<vector_t> &y_test
+  )
+  {
+    const scalar_t epsilon = 0.0000001;
+    int ierr = 0;
+
+    // First compare the norms of the result vector to the baseline
+    Teuchos::Array<scalar_t> norm_test(3);
+    norm_test[0] = y_test->normInf();
+    norm_test[1]=  y_test->norm1();
+    norm_test[2] = y_test->norm2();
+
+    for (size_t i = 0; i < static_cast<size_t>(norm_baseline.size()); i++) {
+      if (std::abs(norm_baseline[i] - norm_test[i]) > epsilon) {
+        ierr++;
+        if (comm->getRank() == 0) {
+          std::cout << "FAIL in test " << testname << ":  norm " << i << ": "
+                    << std::abs(norm_baseline[i] - norm_test[i]) 
+                    << " > " << epsilon
+                    << std::endl;
+          std::cout << "FAIL in test " << testname 
+                    << ": baseline " << norm_baseline[i]
+                    << ": test " << norm_test[i]
+                    << std::endl;
+        }
+      }
+    }
+
+    // If norms match, make sure the vector entries match, too 
+    // (Norms are indifferent to errors in permutation)
+    if (ierr == 0) {
+      y_test->sync_host();
+      yout_baseline->sync_host();
+      auto ytestData = y_test->getLocalViewHost();
+      auto ybaseData = yout_baseline->getLocalViewHost();
+      for (size_t i = 0; i < y_test->getLocalLength(); i++) {
+        if (std::abs(ytestData(i,0) - ybaseData(i,0)) > epsilon) ierr++;
+      }
+      if (ierr > 0) {
+        std::cout << "FAIL in test " << testname << ": " 
+                  << ierr << " vector entries differ on rank "
+                  << comm->getRank() << std::endl;
+      }
+    }
+
+    int gerr = 0;
+    Teuchos::reduceAll<int, int>(*comm, Teuchos::REDUCE_SUM, 1, &ierr, &gerr);
+    return gerr;
+  }
+
+  //////////////////////////////
+  // Run the per-process reader test or the global reader test 
+  int runTest(
+    const std::string &testname,
+    Teuchos::ParameterList &params
+  )
+  {
+    params.set("verbose", true);
+    params.set("callFillComplete", true);
+    params.set("useTimers", true);
+
+    if(perProcess)
+      runTestPerProcess(testname, params);
+    else
+      runTestGlobal(testname, params);
+  }
```

Comment:
Function runTest needs to return the error code from runTestPerProcess or runTestGlobal.

Reply:
Added the missing returns and enabled the tests back. Thanks for the catch! I hope this error was causing the failures.